dockerfiles/Dockerfile.openvino

```diff
@@ -1,92 +1,61 @@
-    apt update && \
-    apt install -y --no-install-recommends ocl-icd-libopencl1 && \
-    rm -rf /var/lib/apt/lists/* && \
-    wget "https://github.com/intel/compute-runtime/releases/download/19.41.14441/intel-gmmlib_19.3.2_amd64.deb" && \
-    wget "https://github.com/intel/compute-runtime/releases/download/19.41.14441/intel-igc-core_1.0.2597_amd64.deb" && \
-    wget "https://github.com/intel/compute-runtime/releases/download/19.41.14441/intel-igc-opencl_1.0.2597_amd64.deb" && \
-    wget "https://github.com/intel/compute-runtime/releases/download/19.41.14441/intel-opencl_19.41.14441_amd64.deb" && \
-    wget "https://github.com/intel/compute-runtime/releases/download/19.41.14441/intel-ocloc_19.41.14441_amd64.deb" && \
-    dpkg -i /tmp/opencl/*.deb && \
-    ldconfig && \
-    rm -rf /tmp/opencl && \
-    cd ${MY_ROOT} && \
-    locale-gen en_US.UTF-8 && update-locale LANG=en_US.UTF-8  && \
-    pip3 install cython && \
-# Download and build ONNX Runtime
-    cd ${MY_ROOT} && \
-    git clone --recursive -b ${ONNXRUNTIME_BRANCH} ${ONNXRUNTIME_REPO} && \
-    /bin/sh onnxruntime/dockerfiles/scripts/install_common_deps.sh && \
-    cd onnxruntime/cmake/external/onnx && python3 setup.py install && \
-    cd ${MY_ROOT}/onnxruntime && ./build.sh --config Release --update --build --parallel --use_openvino ${DEVICE} --build_shared_lib --build_wheel && \
-    pip install build/Linux/Release/dist/*-linux_x86_64.whl && \
-    cd ${MY_ROOT}/ && rm -rf onnxruntime && cd /opt && rm -rf v1.0.22.zip && cd ${MY_ROOT} &&\
-    apt remove -y cmake && cd /usr/share/python-wheels/ && rm -rf *.whl &&\
-    cd /usr/lib/ && rm -rf python2.7 python3.6 python3.8 && cd && rm -rf .cache
+USER root
+RUN apt update; apt install -y git protobuf-compiler libprotobuf-dev
+RUN git clone --recursive -b ${ONNXRUNTIME_BRANCH} ${ONNXRUNTIME_REPO} 
+RUN /bin/sh onnxruntime/dockerfiles/scripts/install_common_deps.sh
+RUN ln -s cmake-* cmake-dir
+RUN python3 -m pip install wheel
+ENV PATH=${WORKDIR_PATH}/cmake-dir/bin:$PATH
+RUN cd onnxruntime/cmake/external/onnx && python3 setup.py install 
+RUN cd onnxruntime && ./build.sh --config Release --update --build --parallel --use_openvino ${DEVICE} --build_shared_lib --build_wheel 
+
+#Steps to download sources
+RUN cat /etc/apt/sources.list | sed 's/^# deb-src/deb-src/g' > ./temp; mv temp /etc/apt/sources.list
+RUN apt update; apt install dpkg-dev
+RUN mkdir /sources
+WORKDIR /sources
+RUN apt-get source cron iso-codes libapt-inst2.0 lsb-release powermgmt-base python-apt-common python3-apt python3-dbus python3-gi unattended-upgrades libapt-pkg5.0 libhogweed4 libnettle6
+WORKDIR /
+RUN tar cvf GPL_sources.tar.gz /sources
+
+# Deploy stage
+FROM openvino/ubuntu18_runtime:${OPENVINO_VERSION}
+
+ENV DEBIAN_FRONTEND noninteractive
+
+USER root
+COPY --from=builder /home/openvino/onnxruntime/build/Linux/Release/dist/*.whl ./
+COPY --from=builder /GPL_sources.tar.gz ./
+RUN python3 -m pip install ./*.whl && rm ./*.whl
```

Comment:
Pip install should be done as USER and not as root. Installing as root will install the package into the root’s python environment and may not be accessible or modify able when running the image as user.  
move this line and preferably the two preceding COPY lines to after the ‘USER ${BUILD_USER}’ line.